src/bin/aperf.rs

```diff
@@ -42,11 +48,19 @@ fn init_logger(verbose: u8) -> Result<()> {
 
 fn main() -> Result<()> {
     let cli = Cli::parse();
+
+    let tmp_dir = TempBuilder::new()
+        .prefix("aperf-tmp-")
+        .tempdir_in(&cli.tmp_dir)?;
+    fs::set_permissions(&tmp_dir, fs::Permissions::from_mode(0o1777))?;
```

Comment:
The previous code was setting only 0o777, not 0o1777. Technically the 1 seems right for a directory, but it wasn't necessary before. Did something change, or did you notice some problem with what was happening before?

Reply:
The 1 sets the sticky bit which is required to prevent users who have access to this dir from deleting the dir.